src/lib/Libutil/misc_utils.c

```diff
@@ -745,32 +744,34 @@ escape_delimiter(char *str, char *delim, char esc, char *retstr, int retstrlen)
 	}
 
 	if ((delim == NULL || *delim == '\0') || esc == '\0') {
-		strcpy(retstr, str);
+		strncpy(retstr, str, strlen(str));
 		return;
 	}
 	delim_len = strlen(delim);
 
 	while (*str != '\0') {
-		if( *str == esc) {
-				retstr[i++] = esc;
-				retstr[i++] = esc;
+		if( *str == esc && i < MAXBUFLEN) {
+				rstr[i++] = esc;
+				rstr[i++] = esc;
 				str++;
-		} else if (strncmp(str, delim, delim_len) == 0) {
-				retstr[i++]=esc;
-			for (j=0; (j < delim_len && i < retstrlen); j++, i++) {
-				retstr[i] = delim[j];
+		} else if(strncmp(str, delim, delim_len) == 0 && i <= MAXBUFLEN) {
+				rstr[i++]=esc;
+			for (j=0; (j < delim_len && i <= MAXBUFLEN); j++, i++) {
+				rstr[i] = delim[j];
 			}
 			str += delim_len;
 		} else {
-			retstr[i++] = *str++;
+			rstr[i++] = *str++;
 		}
 
-		if (i > retstrlen-1) {
+		if (i > MAXBUFLEN) {
```

Comment:
Does this mean that escape_delimiter can not deal with any input string with more than 2048 characters in it?